src/Setup/Agent.php

```diff
@@ -48,6 +48,13 @@ public function getInstallObjective(Config $config = null) : Objective;
      */
     public function getUpdateObjective(Config $config = null) : Objective;
 
+    /**
+     * Get the goal the agent wants to achieve on migration.
+     *
+     * @throw InvalidArgumentException if Config does not match the Agent..
+     */
+    public function getMigrateObjective(Config $config = null): Objective;
```

Comment:
Ich glaube, dass müssten mehrere in einer Liste mit String-Index sein können. Dann kann man die über den Index benennen und addressieren. Das könnte man gut mit den Indizes aus der AgentCollection verbinden. Wenn Filesystem z.B. `["moveFolder"  => ...]` liefert, kann die AgentCollection, die sammelt, `["filesystem.moveFolder" => ...]` haben. Dann können wir den "filesystem.moveFolder" Namen an der Oberfläche in der CLI für den User verwenden. Weiterhin müssten es wohl ein `Migration`-Interface geben, das wahrscheinlich `Objective` erweitern sollte um diese Funktionalitäten für Abschnitte. Vielleicht macht es sogar Sinn, dass das zwei komplett getrennte Interfaces sind, das kann ich gerade noch nicht einschätzen. Vielleicht macht es Sinn, erstmal aufzuschreiben, was eine `Migration` so an Methoden anbieten müsste.

Reply:
Der Name müsste dann auch `getMigrations` sein.

Reply:
ja macht Sinn. Fraglich ist aus meiner Sicht sowieso auch noch, wie lange man Migrations im Source-Code behält, die werden ja irgendwann dann irrelevant. Aber das ist dann eher ein Thema in "von welcher ILIAS Version kann ich wohin direkt upgraden".

Ich würde in dem Fall aber eine Art Collection-Klasse bevorzugen. Arrays als Rückgabewerte (oder Parameter) sind mir eigentlich immer ein Dorn im Auge, da sie eine Konvention innehaben. Mit einer Klasse wird die Konvention forciert. i.O. für dich?